pages/contract-address-calculator/index.page.tsx

```diff
@@ -0,0 +1,102 @@
+import { getContractAddress } from '@ethersproject/address';
+import React, { ReactElement, useState } from 'react';
+
+import { CopyableConversionInput } from '../../src/components/CopyableConversionInput';
+import { CalculatorIcon } from '../../src/components/icons/CalculatorIcon';
+import { Button } from '../../src/components/lib/Button';
+import { Header } from '../../src/components/lib/Header';
+import { NodeBlock } from '../../src/components/lib/NodeBlock';
+import { ToolContainer } from '../../src/components/ToolContainer';
+import { handleChangeValidated } from '../../src/misc/handleChangeValidated';
+import { WithError } from '../../src/misc/types';
+import { addressValidator } from '../../src/misc/validation/validators/addressValidator';
+import { numberValidator } from '../../src/misc/validation/validators/numberValidator';
+
+export default function ContractAddressCalculator(): ReactElement {
+  const [fromAddress, setFromAddress] = useState<WithError<string>>({
+    value: '',
+  });
+  const [nonce, setNonce] = useState<WithError<string>>({
+    value: '',
+  });
+
+  const [calculatedContractAddress, setCalculatedContractAddress] = useState<
+    string | undefined
+  >();
+
+  const flushResults = (): void => setCalculatedContractAddress(undefined);
+
+  const handleChangeFromAddress = (newValue: string): void =>
```

Comment:
There's a convention in this project to use `function` keyword instead of arrow functions.

Reply:
I will do this but notice how `calldata-decoder` also uses arrow functions(where I copied boilerplate from)

Reply:
Oh, sorry then. Anyways - we should follow that convention. I will update the functions in the other places then. Thanks!